erpnext/accounts/doctype/gl_entry/gl_entry.py

```diff
@@ -97,6 +100,30 @@ def validate_account_details(self, adv_adj):
 			frappe.throw(_("{0} {1}: Account {2} does not belong to Company {3}")
 				.format(self.voucher_type, self.voucher_no, self.account, self.company))
 
+	def validate_accounting_period(self):
```

Comment:
This function should be added in `general_ledger.py` and should only be **called once** for a transaction.